naturalcrit/homebrewery
# adjust frontend error handling, add client/server mismatch middleware

Testing this is rather simple! 
1. Navigate to the home, new, or edit page.
2. Open the js console, and set `version` to anything that isn't the current version.
3. Trigger a save.
4. The error should appear, explaining what's wrong.

The logic is also quite simple. 
* Whenever an http call is made to the API, a superagent middleware pulls the version out of `global`, and places it on a header called `Homebrewery-Version`.
* The API now has a middleware that checks the version in that header against the server's version. If they don't match, it returns a 412 error.
  * I checked all the http response codes and couldn't find one that matched exactly, but 412 seemed reasonable based off its definition.

~At this time this is not complete, I still need to finish the brews page and metadata editor's deletion error handling.~
